### DISCUSSION TITLE
A model can train and evalutate, but optuna study caused OOM after several trials

### DISCUSSION THREAD

**Discussion by EvoLandEco** [optuna/optuna#5136] [Q&A]
Hi all, I am using PyTorch Geometric to train graph neural network models.

Half month ago, I can use Optuna without a problem to do a 48-Hour study, with around 150+ trials. Yesterday I tried Optuna again on the same model, same dataset, same batch size and same device (A100 40GB or V100 32GB), but I always got `torch.cuda.OutOfMemoryError: CUDA out of memory` after around 16 trials.

I enabled GC by using `study.optimize(objective, timeout=172800, gc_after_trial=True`, I also tried to add `gc.collect()` and `cuda.empty_cache()` after each epoch, but this didn't help. I even tried to largely reduce the scales of the hidden layers, this also didn't help.

I would like to know if there are any general approaches to fully release VRAM after each trial? Or, is there any method to avoid OOM? It is really weird because my first use of Optuna was just fine, I might have changed some parameters in the model, but I am really a newbie in machine learning, that I have no idea where to start to look for potential issues. Would be great if anyone could shed some light on the largest factors impacting VRAM and RAM use when training and optimizing a GNN model.

**Comment by nzw0301** [comment]
Could you share the minimal reproducible code with us when you ask a question? Otherwise, others hardly help you resolve the issue.
